src/v/net/unresolved_address.h

```diff
@@ -44,8 +44,7 @@ class unresolved_address
     inet_family family() const { return _family; }
 
     bool operator==(const unresolved_address& other) const = default;
-    friend bool operator<(const unresolved_address&, const unresolved_address&)
-      = default;
+    std::strong_ordering operator<=>(const unresolved_address&) const = default;
```

Comment:
nit: once there is a defaulted <=> a separate default == it's not necessary

Reply:
mm, i don't think that spaceship operator covers ==?

Reply:
it does https://gcc.godbolt.org/z/943qon97o (reusing an earlier example)
There could be reasons to have a defaulted spaceship and a user-provided ==, for example, if the programmer wants to do something faster for == than what the default would provide. But if both are defaulted, it's redundant.